bot/decorators.py

```diff
@@ -45,6 +45,45 @@ def predicate(ctx: Context) -> bool:
     return commands.check(predicate)
 
 
+def not_in_blacklist(
+    *,
+    channels: t.Container[int] = (),
+    categories: t.Container[int] = (),
+    roles: t.Container[int] = (),
+    override_roles: t.Container[int] = (),
+    redirect: t.Optional[int] = Channels.bot_commands,
+    fail_silently: bool = False,
+) -> t.Callable:
+    """
+    Check if a command was not issued in a blacklisted context.
+
+    The blacklists that can be provided are:
+
+    - `channels`: a container with channel ids for blacklisted channels
+    - `categories`: a container with category ids for blacklisted categories
+    - `roles`: a container with role ids for blacklisted roles
+
+    If the command was invoked in a context that was blacklisted, the member is either
+    redirected to the `redirect` channel that was passed (default: #bot-commands) or simply
+    told that they're not allowed to use this particular command (if `None` was passed).
+
+    The blacklist can be overridden through the roles specified in `override_roles`.
+    """
+    def predicate(ctx: Context) -> bool:
+        """Check if command was issued in a blacklisted context."""
+        not_blacklisted = not in_whitelist_check(ctx, channels, categories, roles, fail_silently=True)
+        overridden = in_whitelist_check(ctx, roles=override_roles, fail_silently=True)
```

Comment:
What do you think about making this a separate decorator like `@override_blacklist`? It would work by catching the `InWhitelistCheckFailure` raised and suppresing it. I feel like shoving so much into a single decorator makes the API confusing.

Reply:
I think it sound ok, I just need to figure out how to do it lol

Reply:
It seems this hasn't been addressed. I have no opinion on the matter, but everything else looks good to me.

Reply:
@HassanAbouelela Yeah, we talked about it in the server and decided it's more trouble than it's worth, since checks are handled like normal decorators. This would mean this would need to be either rewritten as non-dpy checks, or monkey-patch an attribute to be inspected in another part of the code.

Reply:
For sir-lancebot, we do monkey patching like [this](https://github.com/python-discord/sir-lancebot/blob/1e9a88f5b59cf7d1c4e24f48cc09aa98e5f6499b/bot/utils/decorators.py#L206), and it isn't too bad. I do agree this isn't necessary here, as blacklist is not a global check, and can be directly modified by the caller.